drivers/gpio/gpio_pcf8575.c

```diff
@@ -0,0 +1,410 @@
+/**
+ * Copyright (c)
+ * 2022 Ithinx GmbH
+ * 2023 Mr Beam Lasers GmbH
+ * 
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * @see https://www.nxp.com/docs/en/data-sheet/PCF8575.pdf
+ */
+
+#define DT_DRV_COMPAT nxp_pcf8575
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pcf8575, CONFIG_GPIO_LOG_LEVEL);
+
+struct pcf8575_pins_cfg {
+	uint16_t configured_as_outputs; /* 0 for input, 1 for output */
+	uint16_t outputs_state;
+};
+
+/** Runtime driver data of the pcf8575*/
+struct pcf8575_drv_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+	struct pcf8575_pins_cfg pins_cfg;
+	sys_slist_t callbacks;
+	struct k_sem lock;
+	struct k_work work;
+	const struct device *dev;
+	struct gpio_callback int_gpio_cb;
+	uint16_t input_port_last;
+};
+
+/** Configuration data*/
+struct pcf8575_drv_cfg {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	struct i2c_dt_spec i2c;
+	struct gpio_dt_spec gpio_int;
+};
+
+/**
+ * @brief Reads the value of the pins from pcf8575 respectively from a connected device.
+ *
+ * @param dev Pointer to the device structure of the driver instance.
+ * @param value Pointer to the input value. It contains the received Bytes(receives 2 Bytes for P0 and P1).
+ *
+ * @retval 0 If successful.
+ * @retval Negative value for error code.
+ */
+static int pcf8575_process_input(const struct device *dev, gpio_port_value_t *value)
+{
+	const struct pcf8575_drv_cfg *drv_cfg = dev->config;
+	struct pcf8575_drv_data *drv_data = dev->data;
+	int rc = 0;
+	uint16_t rx_buf;
+	uint8_t rx_buf_helper[2];
+	
+	rc = i2c_read_dt(&drv_cfg->i2c, rx_buf_helper, sizeof(rx_buf_helper));
+	if (rc != 0) {
+		LOG_ERR("%s: failed to read from device: %d", dev->name, rc);
+		return -EIO;
```

Comment:
what is "-EIO" ?

Reply:
#define	EIO 5		/* I/O error */ from errno.h